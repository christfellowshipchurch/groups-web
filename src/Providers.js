import React from 'react';
import { ThemeProvider } from 'styled-components';
import PropTypes from 'prop-types';

import { defaultTheme } from './ui-kit/theme';

const Providers = ({ children }) => (
  <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
);

Providers.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default Providers;
