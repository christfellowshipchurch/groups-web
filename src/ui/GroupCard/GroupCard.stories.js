import React from 'react';
// import { storiesOf } from '@storybook/react';

import GroupCard from '.';

const Template = (args) => <GroupCard {...args} />;

const Story = {
  component: GroupCard,
  title: 'ui/GroupCard',
  args: {
    coverImage: 'https://picsum.photos/840/472',
    title: 'Group Name',
  },
};

const example = Template.bind({});
example.args = {
  groupSize: 11,
  label: 'Label',
  location: 'Location',
};

const groupSize = Template.bind({});
groupSize.storyName = 'groupSize';
groupSize.args = {
  groupSize: 11,
};

const isLoading = Template.bind({});
isLoading.storyName = 'isLoading';
isLoading.args = {
  groupSize: 11,
  isLoading: true,
  label: 'Label',
  location: 'Location',
};

const label = Template.bind({});
label.storyName = 'label';
label.args = {
  label: 'Label',
};

const location = Template.bind({});
location.storyName = 'location';
location.args = {
  location: 'Location',
};

export { Story as default, example, groupSize, isLoading, label, location };
