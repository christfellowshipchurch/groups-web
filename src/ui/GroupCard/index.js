import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import { Groups, Pin } from '../../ui-kit/theme/icons';
import { CardContent, CardWrapper, H4, H6 } from '../../ui-kit';

const Card = styled(CardWrapper)`
  min-width: calc(320px - ${({ theme }) => theme.sizing.baseUnit(2)});
  max-width: 420px;
  overflow: hidden;
`;

const CardLabel = styled(H4)`
  position: absolute;
  top: ${({ theme }) => theme.sizing.baseUnit(-2.3125)};
  left: 0;
  background: ${({ theme }) => theme.colors.text.primary};
  color: white;
  padding: ${({ theme }) =>
    `${theme.sizing.baseUnit(0.3125)} ${theme.sizing.baseUnit(1.25)}`};
`;

const GroupDetails = styled.hgroup`
  display: flex;
  flex-direction: row;
  align-items: center;

  & > h6 {
    margin: ${({ theme }) => `0 ${theme.sizing.baseUnit(0.3125)} 0 0`};
  }
`;

const iconStyles = css`
  fill: ${({ theme }) => theme.colors.primary};
  height: 11px;
  margin: ${({ theme }) =>
    `0 ${theme.sizing.baseUnit(0.125)} ${theme.sizing.baseUnit(0.125)} 0`};
  width: auto;
`;

const LocationIcon = styled(Pin)`
  ${iconStyles}
`;

const GroupSizeIcon = styled(Groups)`
  ${iconStyles}
`;

function GroupCard({ image, label, location, groupName, groupSize }) {
  return (
    <>
      <Card>
        <img src={image} width="100%" height="auto" alt="Default Card Text" />
        <CardContent style={{ position: 'relative' }}>
          <CardLabel>{label}</CardLabel>
          <H4>{groupName}</H4>
          <GroupDetails>
            <LocationIcon />
            <H6>{location}</H6>
            <GroupSizeIcon />
            <H6>{groupSize}</H6>
          </GroupDetails>
        </CardContent>
      </Card>
    </>
  );
}

GroupCard.proptypes = {
  image: PropTypes.string,
  label: PropTypes.string,
  location: PropTypes.string,
  name: PropTypes.string,
  size: PropTypes.string,
};

export default GroupCard;
