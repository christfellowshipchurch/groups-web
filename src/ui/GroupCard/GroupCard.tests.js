import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Providers from '../../Providers';

import GroupCard from '.';

describe('the GroupCard component', () => {
  it('should render', () => {
    const tree = renderer.create(
      <Providers>
        <GroupCard
          coverImage={'https://picsum.photos/840/472'}
          title={'Group Name'}
        />
      </Providers>
    );

    expect(tree).toMatchSnapshot();
  });
  it('should render with a groupSize', () => {
    const tree = renderer.create(
      <Providers>
        <GroupCard
          coverImage={'https://picsum.photos/840/472'}
          groupSize={11}
          title={'Group Name'}
        />
      </Providers>
    );

    expect(tree).toMatchSnapshot();
  });
  it('should render a loading state ', () => {
    const tree = renderer.create(
      <Providers>
        <GroupCard
          coverImage={'https://picsum.photos/840/472'}
          groupSize={11}
          isLoading
          label={'Label Name'}
          location={'Location Name'}
          title={'Group Name'}
        />
      </Providers>
    );

    expect(tree).toMatchSnapshot();
  });
  it('should render with a label', () => {
    const tree = renderer.create(
      <Providers>
        <GroupCard
          coverImage={'https://picsum.photos/840/472'}
          label={'Label Name'}
          title={'Group Name'}
        />
      </Providers>
    );

    expect(tree).toMatchSnapshot();
  });
  it('should render with a location', () => {
    const tree = renderer.create(
      <Providers>
        <GroupCard
          coverImage={'https://picsum.photos/840/472'}
          location={'Location Name'}
          title={'Group Name'}
        />
      </Providers>
    );

    expect(tree).toMatchSnapshot();
  });
});
