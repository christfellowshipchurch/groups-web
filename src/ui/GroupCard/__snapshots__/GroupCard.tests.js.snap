// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the GroupCard component should render 1`] = `
.c3 {
  padding: 1rem 1rem;
  padding: 1.25rem;
}

.c2 {
  width: 100%;
  height: auto;
}

.c0 {
  background: #FFFFFF;
  border-radius: 5px;
  box-shadow: 0 10px 9px -4px rgba(0,0,0,.07);
  margin: 0.75rem 1rem;
}

.c4 {
  font-size: 1rem;
  font-weight: 700;
  line-height: 1.5rem;
  margin: initial;
}

.c1 {
  max-width: 420px;
  overflow: hidden;
}

<div
  className="c0 c1 o4tkys-0 c1"
>
  <img
    alt="Group Name"
    className="c2 sc-1shebyi-1 kaSgGP"
    src="https://picsum.photos/840/472"
  />
  <div
    className="sc-9taxng-0 c3"
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <h4
      className="c4 sc-17mtry8-2 bKsXwZ"
    >
      Group Name
    </h4>
  </div>
</div>
`;

exports[`the GroupCard component should render a loading state  1`] = `
.c5 {
  background: #A5A5A5;
  border-radius: 5px;
  display: block;
  height: 1rem;
}

.c2 {
  background: #A5A5A5;
  border-radius: 5px;
  display: block;
  width: 100%;
  padding-top: 100%;
}

.c8 {
  background: #A5A5A5;
  border-radius: 5px;
  display: block;
  width: 0.6875rem;
  padding-top: 0.6875rem;
}

.c4 {
  padding: 1rem 1rem;
  padding: 1.25rem;
}

.c3 {
  border-radius: 5px 5px 0 0;
}

.c0 {
  background: #FFFFFF;
  border-radius: 5px;
  box-shadow: 0 10px 9px -4px rgba(0,0,0,.07);
  margin: 0.75rem 1rem;
}

.c6 {
  height: 1rem;
  margin: calc(1.5rem - 1rem) 0;
}

.c9 {
  height: 0.75rem;
  margin: 0 0 calc(1.125rem - 0.75rem) 0;
}

.c1 {
  max-width: 420px;
  overflow: hidden;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 > h6 {
  margin: 0 0.3125rem 0 0;
}

.c7 > .custom-placeholder {
  margin: 0 0.3125rem 0 0;
}

.c7 > .custom-placeholder.line {
  width: 15%;
  margin-bottom: 0;
}

<div
  className="c0 c1 o4tkys-0 c1"
>
  <mark
    className="c2 show-loading-animation c3 media custom-placeholder"
    size="100%"
  />
  <div
    className="sc-9taxng-0 c4"
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <mark
      className="c5 show-loading-animation ra2rjz-1 c6 line custom-placeholder"
    />
    <mark
      className="c5 show-loading-animation ra2rjz-1 c6 line custom-placeholder"
    />
    <hgroup
      className="c7"
    >
      <mark
        className="c8 media custom-placeholder"
        size={0.6875}
      />
      <mark
        className="c5 show-loading-animation ra2rjz-1 c9 line custom-placeholder"
      />
      <mark
        className="c8 media custom-placeholder"
        size={0.6875}
      />
      <mark
        className="c5 show-loading-animation ra2rjz-1 c9 line custom-placeholder"
      />
    </hgroup>
  </div>
</div>
`;

exports[`the GroupCard component should render with a groupSize 1`] = `
.c6 {
  fill: #303030;
  height: auto;
  width: 0.6875rem;
}

.c3 {
  padding: 1rem 1rem;
  padding: 1.25rem;
}

.c2 {
  width: 100%;
  height: auto;
}

.c0 {
  background: #FFFFFF;
  border-radius: 5px;
  box-shadow: 0 10px 9px -4px rgba(0,0,0,.07);
  margin: 0.75rem 1rem;
}

.c4 {
  font-size: 1rem;
  font-weight: 700;
  line-height: 1.5rem;
  margin: initial;
}

.c8 {
  color: #505050;
  font-size: 0.75rem;
  font-weight: 500;
  line-height: 1.125rem;
  margin: initial;
}

.c1 {
  max-width: 420px;
  overflow: hidden;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 > h6 {
  margin: 0 0.3125rem 0 0;
}

.c5 > .custom-placeholder {
  margin: 0 0.3125rem 0 0;
}

.c5 > .custom-placeholder.line {
  width: 15%;
  margin-bottom: 0;
}

.c7 {
  fill: salmon;
  margin: 0 0.125rem 0.125rem 0;
}

<div
  className="c0 c1 o4tkys-0 c1"
>
  <img
    alt="Group Name"
    className="c2 sc-1shebyi-1 kaSgGP"
    src="https://picsum.photos/840/472"
  />
  <div
    className="sc-9taxng-0 c3"
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <h4
      className="c4 sc-17mtry8-2 bKsXwZ"
    >
      Group Name
    </h4>
    <hgroup
      className="c5"
    >
      <Svgr
        className="c6 c7"
      />
      <h6
        className="c8 toneu3-2 ctXJce"
      >
        11
      </h6>
    </hgroup>
  </div>
</div>
`;

exports[`the GroupCard component should render with a label 1`] = `
.c3 {
  padding: 1rem 1rem;
  padding: 1.25rem;
}

.c2 {
  width: 100%;
  height: auto;
}

.c0 {
  background: #FFFFFF;
  border-radius: 5px;
  box-shadow: 0 10px 9px -4px rgba(0,0,0,.07);
  margin: 0.75rem 1rem;
}

.c4 {
  font-size: 1rem;
  font-weight: 700;
  line-height: 1.5rem;
  margin: initial;
}

.c1 {
  max-width: 420px;
  overflow: hidden;
}

.c5 {
  position: absolute;
  top: -2.125rem;
  left: 0;
  background: #303030;
  color: white;
  padding: 0.3125rem 1.25rem;
}

<div
  className="c0 c1 o4tkys-0 c1"
>
  <img
    alt="Group Name"
    className="c2 sc-1shebyi-1 kaSgGP"
    src="https://picsum.photos/840/472"
  />
  <div
    className="sc-9taxng-0 c3"
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <h4
      className="c4 c5 sc-17mtry8-2 o4tkys-1 c5"
    >
      Label Name
    </h4>
    <h4
      className="c4 sc-17mtry8-2 bKsXwZ"
    >
      Group Name
    </h4>
  </div>
</div>
`;

exports[`the GroupCard component should render with a location 1`] = `
.c6 {
  fill: #303030;
  height: auto;
  width: 0.6875rem;
}

.c3 {
  padding: 1rem 1rem;
  padding: 1.25rem;
}

.c2 {
  width: 100%;
  height: auto;
}

.c0 {
  background: #FFFFFF;
  border-radius: 5px;
  box-shadow: 0 10px 9px -4px rgba(0,0,0,.07);
  margin: 0.75rem 1rem;
}

.c4 {
  font-size: 1rem;
  font-weight: 700;
  line-height: 1.5rem;
  margin: initial;
}

.c8 {
  color: #505050;
  font-size: 0.75rem;
  font-weight: 500;
  line-height: 1.125rem;
  margin: initial;
}

.c1 {
  max-width: 420px;
  overflow: hidden;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 > h6 {
  margin: 0 0.3125rem 0 0;
}

.c5 > .custom-placeholder {
  margin: 0 0.3125rem 0 0;
}

.c5 > .custom-placeholder.line {
  width: 15%;
  margin-bottom: 0;
}

.c7 {
  fill: salmon;
  margin: 0 0.125rem 0.125rem 0;
}

<div
  className="c0 c1 o4tkys-0 c1"
>
  <img
    alt="Group Name"
    className="c2 sc-1shebyi-1 kaSgGP"
    src="https://picsum.photos/840/472"
  />
  <div
    className="sc-9taxng-0 c3"
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <h4
      className="c4 sc-17mtry8-2 bKsXwZ"
    >
      Group Name
    </h4>
    <hgroup
      className="c5"
    >
      <Svgr
        className="c6 c7"
      />
      <h6
        className="c8 toneu3-2 ctXJce"
      >
        Location Name
      </h6>
    </hgroup>
  </div>
</div>
`;
